### PR TITLE
feat(FEC-11875): Related Entries - by context

### PR DESCRIPTION
**Pull request by SivanA-Kaltura**
### Description of the Changes

Update player type signature to support usage in related entries plugin

Resolves FEC-11875
Related PRs: 
https://github.com/kaltura/playkit-js-related/pull/4
https://github.com/kaltura/playkit-js-providers/pull/171

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Review by yairans — DISMISSED** [review]

**Review by yairans — APPROVED** [review]

**Review by yairans — APPROVED** [review]

File: ts-typed/player.d.ts [hunk 1]
```diff
@@ -24,7 +24,8 @@ declare namespace KalturaPlayerTypes {
     paused: boolean;
     seeking: boolean;
     isOnLiveEdge: () => boolean;
-    loadMedia: (options: {entryId: string; ks?: string}) => void;
+    loadMedia: (options: {entryId: string; ks?: string}) => Promise<any>;
```
Comment (yairans) [inline ts-typed/player.d.ts:27]:
```suggestion
    loadMedia: (mediaInfo: OVPMediaInfo|OTTMediaInfo, mediaOptions: any) => Promise<any>;
```

**Review by yairans — CHANGES_REQUESTED** [review]

**Review by yairans — CHANGES_REQUESTED** [review]

File: ts-typed/player-config.d.ts [hunk 2]
```diff
@@ -187,4 +186,22 @@ declare namespace KalturaPlayerTypes {
     debug?: boolean;
     log?: Log;
   }
+
```
Comment (yairans) [inline ts-typed/player-config.d.ts:189]: Lets create a generic type which is OVPMediaInfo|OTTMediaInfo to keep the provider type transparent

**Review by yairans — CHANGES_REQUESTED** [review]

File: ts-typed/player-config.d.ts (hunk 2 above)
Comment (SivanA-Kaltura) [reply ts-typed/player-config.d.ts:189]: From what I saw we can't do it in the same way the flow definition does.
What i can do it make interface MediaInfo and have 2 separate interfaces extending it.

Comment (yairans) [reply ts-typed/player-config.d.ts:189]: ok

**Review by yairans — APPROVED** [review]

File: ts-typed/player-config.d.ts [hunk 3]
```diff
@@ -27,8 +27,7 @@ declare namespace KalturaPlayerTypes {
   }
 
   export enum EntryTypes {
-    Vod = 'Vod',
-    Live = 'Live',
+    Vod = 'Vod'
```
Comment (yairans) [inline ts-typed/player-config.d.ts:30]: why Live has been removed?

**Pull request merged**